### PR TITLE
[APPC-1740][APPC-1746] Pay as Guest - Check for ongoing transactions

### PR DESCRIPTION
**Pull request by JDSM01**
**What does this PR do?**

Adds a check for ongoing transactions before loading the payment methodss
Fixes bug on rotation while loading payment methods

**Database changed?**

   No

**Where should the reviewer start?**

PaymentMethodsPresenter.java

**How should this be manually tested?**

Be very fast and kill the app before the transaction completes and try to make another purchase and check if the it retakes the previous transaction.
Use charles to mock transaction (you need to have done a transaction before, otherwise it will not return a purchase after) to return PROCESSING and check if it redirects and completes the purchase.

**What are the relevant tickets?**

https://aptoide.atlassian.net/browse/APPC-1740
**Questions:**

   Does this add new dependencies which need to be added? (Eg. new libs, new keys, etc.) 

**Code Review Checklist**

- [ ] Architecture
- [ ] Documentation on public interfaces
- [ ] Database changed?
- [ ] If yes - Database migration?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] Functional QA tests pass

### CODE REVIEW

File: android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/mappers/PurchaseMapper.java [hunk 1]
```diff
@@ -41,13 +41,9 @@ public PurchasesModel mapList(RequestResponse requestResponse) {
       try {
         JSONArray jsonArray = new JSONObject(response).optJSONArray("items");
         for (int i = 0; i < jsonArray.length(); i++) {
-          try {
-            JSONObject jsonObject = jsonArray.getJSONObject(i);
-            SkuPurchase skuPurchase = mapSkuPurchase(jsonObject);
-            skuPurchases.add(skuPurchase);
-          } catch (Exception e) {
```
Comment (Trinkes) [inline android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/mappers/PurchaseMapper.java:48]: why did you remove the try catch?

File: android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/models/Transaction.java [hunk 2]
```diff
@@ -2,6 +2,50 @@
 
 public class Transaction {
 
+  private final String uid;
+  private final String hash;
+  private final String orderReference;
+  private final String status;
```
Comment (Trinkes) [inline android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/models/Transaction.java]: why don't you use Status enum type instead of string?

File: android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/models/Transaction.java [hunk 3]
```diff
@@ -2,6 +2,50 @@
...
+      String gateway) {
+
+    this.uid = uid;
+    this.hash = hash;
+    this.orderReference = orderReference;
+    this.status = status;
+    this.gateway = gateway;
+  }
+
+  public Transaction() {
```
Comment (Trinkes) [inline android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/models/Transaction.java]: Did you consider creating a static method to create a `FailedTransaction` instead of a constructor with no parameters?

File: android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/payasguest/AdyenPaymentFragment.java [hunk 4]
```diff
@@ -514,6 +518,13 @@ private String getBundleString(String key) {
     throw new IllegalArgumentException(key + "data not found");
   }
 
+  private boolean getBundleBoolean(String key) {
```
Comment (Trinkes) [inline android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/payasguest/AdyenPaymentFragment.java]: not used

File: android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/payasguest/IabView.java [hunk 5]
```diff
@@ -15,8 +15,11 @@
 
   void redirectToWalletInstallation(Intent intent);
 
+  /**
+   * Pass a non-empty uid if you want to resume a transaction. Otherwise send null or empty
+   */
```
Comment (Trinkes) [inline android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/payasguest/IabView.java]: I think you should create 2 different methods: 1 to resume a purchase, other to start it instead of using a null argument

**Review by Trinkes — CHANGES_REQUESTED** [review]

File: android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/mappers/PurchaseMapper.java (hunk 1 above)
Comment (JDSM01) [reply android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/mappers/PurchaseMapper.java:48]: Read it wrong at the time, you're right, it shouldn't have been removed

File: android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/models/Transaction.java (hunk 3 above)
Comment (JDSM01) [reply android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/models/Transaction.java]: So that's how you create an empty constructor with a verbose name. But FailedTransaction is not the best name. ErrorTransaction is a better suited name I think

Comment (Trinkes) [reply android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/models/Transaction.java]: Agree, i just wanted to explain the idea

File: android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/utils/EnumMapper.java [hunk 6]
```diff
@@ -0,0 +1,10 @@
+package com.appcoins.sdk.billing.utils;
+
+public class EnumMapper {
+
+  public Enum parseToEnum(Class enumClass, String string) {
```
Comment (Trinkes) [inline android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/utils/EnumMapper.java]: can we have a default value for the cases where we don't know the value?for instance if we add a new status

**Review by Trinkes — CHANGES_REQUESTED** [review]
